src/library/basetracktablemodel.cpp

```diff
@@ -474,6 +478,37 @@ QVariant BaseTrackTableModel::roleValue(
     case Qt::ToolTipRole:
         if (column == fieldIndex(ColumnCache::COLUMN_LIBRARYTABLE_COLOR)) {
             return mixxx::RgbColor::toQString(mixxx::RgbColor::fromQVariant(rawValue));
+        } else if (column == fieldIndex(ColumnCache::COLUMN_LIBRARYTABLE_COVERART)) {
+            // Determine height of the cover art image depending on the screen size
+            unsigned int absHeightOfCoverartTooltip;
+            const QScreen* primaryScreen = getPrimaryScreen();
+            if (primaryScreen) {
+                absHeightOfCoverartTooltip = static_cast<int>(
+                        primaryScreen->availableGeometry().height() *
+                        kRelHeightOfCoverartTooltip);
+            } else {
+                qWarning() << "Assuming screen height of 800px.";
+                absHeightOfCoverartTooltip = 800 * kRelHeightOfCoverartTooltip;
+            }
+            // Get image from cover art cache
+            CoverArtCache* pCache = CoverArtCache::instance();
+            QPixmap pixmap = QPixmap(absHeightOfCoverartTooltip,
+                    absHeightOfCoverartTooltip); // Height also used as default for the width, in assumption that covers are squares
+            pixmap = pCache->tryLoadCover(this,
+                    getCoverInfo(index),
+                    absHeightOfCoverartTooltip,
+                    CoverArtCache::Loading::NoSignal);
+            if (pixmap.isNull()) {
+                // Cache miss -> Don't show a tooltip
+                return kEmptyString;
+            }
+            QByteArray data;
+            QBuffer buffer(&data);
+            pixmap.save(&buffer, "PPM"); // Most simple format, without compression effort
+            QString html = QString(
```

Comment:
Use QStringLiteral and operator + from QStringBuilder to compose this string.

Did you try to return the QPixmap directly as a variant? Not sure if this might work.

Reply:
It compiles, but no tooltip is shown.

Reply:
PPM is very inefficient and bloated. Plain ASCII text that get expanded even more by the base64 encoding.

Use "PNG" with quality=100 to disable compression and leave a comment.

Reply:
The PNG  format contains a Lossy and a Lossless compression. I guess that 100% quality does not disable the calculation overhead of the lossless compression.
What do you think about the BMP format, which is a more dense binary format than PPM, but also without compression overhead?

Reply:
Any uncompressed, binary format is suitable.

Reply:
Changed format to BMP